### ISSUE TITLE
Denied Kubernetes interactive sessions do not return the error to client

### ISSUE THREAD

**Issue by tigrato** [gravitational/teleport#13023]
Expected behavior:
It is expected that any interactive `kubectl exec -it`  command that Kubernetes RBAC denies fails and returns the error to the user with the correct exit code.

Current behavior:
Interactive sessions return nothing to the user when they are denied and the exit code is always  `0`.

Bug details:

Given the following teleport user:
```
kind: role
version: v3
metadata:
  name: contractor
spec:
  allow:
    kubernetes_groups:
    - z4937-manual
    request:
      roles: ['dba']
---
kind: user
metadata:
  name: z4937m
spec:
  roles:
  - contractor
  status:
    is_locked: false
    lock_expires: "0001-01-01T00:00:00Z"
    locked_time: "0001-01-01T00:00:00Z"
  traits:
    logins:
    - z4937m
version: v2
```
and the following Kubernetes objects:
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: z4937-manual
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: z4937-manual
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: Group
  name: z4937-manual
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: z4937-manual
rules:
- apiGroups:
  - ""
  resources:
  - users
  - groups
  - serviceaccounts
  verbs:
  - impersonate
- apiGroups:
  - ""
  resources:
  - pods
  verbs:
  - get
  - list
- apiGroups:
  - authorization.k8s.io
  resources:
  - selfsubjectaccessreviews
  verbs:
  - create
```

It is expected that any `kubectl exec -it`  fails and returns the error to the user.

The error is correctly returned by Kubernetes and caught by Teleport but it's never delivered to the user if the session is interactive.

https://github.com/gravitational/teleport/blob/ce4d6234ff29d96a63bedd4df55b2981e6680097/lib/kube/proxy/sess.go#L861

The user connection is already upgraded and we cannot write into the response writer.

Version: any  9.x.x

initially reported by @programmerq  on #12283

gz#5254

**Comment by programmerq** [comment]
@tigrato The customer in question is interested in a backport to the v9 branch for the issue described in both this and #12283. Are those planned?

Thanks!

**Comment by tigrato** [comment]
> @tigrato The customer in question is interested in a backport to the v9 branch for the issue described in both this and #12283. Are those planned?
> 
> Thanks!

Yes. I wrote a fix for this issue and #12283 can also be backported.

@programmerq, if you have time, can you check if the PR fixes the behavior on your side?

**Comment by tigrato** [comment]
@programmerq both fixes were backported to v9
